dmn/src/main/java/io/camunda/zeebe/dmn/DecisionResult.java

```diff
@@ -31,4 +32,13 @@
 
   /** @return the output of the decision if it was made successfully, otherwise {@code null} */
   DirectBuffer getOutput();
+
+  /**
+   * Returns the details of the evaluated decision and its required decisions. The order depends on
+   * the evaluation order, starting from the required decisions. If the evaluation is not successful
+   * then it contains the successful evaluated decisions and the decision that was not successful.
```

Comment:
🔧 Since nothing is controlling the evaluation order in zeebe, we should write a test for it that should fail as soon as the evaluation order changes and this documentation is no longer valid.

Reply:
Yes. The order is verified in the test case [DmnEvaluatedDecisionsTest#shouldReturnResultOfAllEvaluatedDecisions](https://github.com/camunda-cloud/zeebe/pull/8734/files#diff-5a9b6f3ae4beceeef6fc9874d1ab893238a0dc976232c4d8101df2d15c06726fR108-R113). I'm adding additional description to make the expectation more clear.

Reply:
I see, but it only covers a simple first, second scenario. I think what we need here is to verify the order of results in a more complex DRG, where most decisions require multiple other decisions.

Reply:
Yes, it would be good to have a more complex example. Let's do something in the future